src/analysis/excluded_terms_in_mondo/excluded_terms_in_mondo.py

```diff
@@ -0,0 +1,167 @@
+"""Excluded terms in / xrefed-in Mondo
+
+Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/31
+
+TODO's
+"""
+import os
+import sys
+from argparse import ArgumentParser
+from typing import Dict, List
+
+import pandas as pd
+
+
+# Vars
+# # Static
+THIS_SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+# CACHE_DIR = os.path.join(THIS_SCRIPT_DIR, 'cache')
+ANALYSIS_DIR = os.path.join(THIS_SCRIPT_DIR, '..')
+PROJECT_DIR = os.path.join(ANALYSIS_DIR, '..', '..')
+MAPPINGS_DIR = os.path.join(THIS_SCRIPT_DIR, 'ignored', 'mappings')
+SIGNATURE_FILES_DIR = os.path.join(PROJECT_DIR, 'src', 'ontology', 'reports')
+# # Config
+CONFIG = {
+    'ontologies': ['doid', 'icd10cm', 'icd10who', 'ncit', 'omim', 'ordo'],  # todo: get list from mondo config yml
+    # 'use_cache': False,
+    'mondo_mappings_path': os.path.join(MAPPINGS_DIR, 'mondo.sssom.tsv'),
+    'component_sig_pattern': 'component_signature-{}.tsv',
+    'mirror_sig_pattern': 'mirror_signature-{}.tsv',
+    'outpath_xrefed_terms': os.path.join(PROJECT_DIR, 'excluded_terms_in_mondo - xrefs.csv'),
+    'outpath_in_terms': os.path.join(PROJECT_DIR, 'excluded_terms_in_mondo - in_mondo.csv'),
+    'save': True,  # save dataframes to disk
+
+}
+
+
+# Functions
+def uri_to_curie(uri: str) -> str:
+    """Takes an ontological URI and returns a CURIE. Works on the following patterns:
+    - http://.../PREFIX/CODE
+    - http://.../PREFIX_CODE"""
+    uri_list: List[str] = uri.split('/')
+    end_bit = uri_list[-1]
+    if '_' in end_bit:
+        curie = end_bit.replace('_', ':')
+    else:
+        curie = uri_list[-2] + ':' + end_bit
+    return curie
+
+
+def load_and_format_tsv(onto_name: str, pattern: str, _dir: str = SIGNATURE_FILES_DIR) -> pd.DataFrame:
+    """Load and format"""
+    # Get path
+    path = os.path.join(_dir, pattern.format(onto_name))
+    # Read
+    df = pd.read_csv(path, sep='\t').fillna('')
+    # Format
+    df['?term'] = df['?term'].apply(lambda x: x.strip())
+    df['?term'] = df['?term'].apply(uri_to_curie)
+    df = df.rename({'?term': 'term_id'})
+    return df
+
+
+# TODO
+def run_xrefed(mirror_df: pd.DataFrame, component_df: pd.DataFrame, mondo_mappings: pd.DataFrame) -> pd.DataFrame:
+    """Run for a single ontology"""
+    # TODO: Find prefix. make map between onto names and prefix? DOID, OMIM, OMIMPS, Orphanet, NCIT, prolly ICD10CM ICD10WHO
+    # TODO: More stable to sample a term URI base and get prefix from prefix map
+    df = pd.DataFrame()
+    return df
+
+
+# TODO
+def run(
+    ontologies: List[str], mondo_mappings_path: str, component_sig_pattern: str, mirror_sig_pattern: str,
+    outpath_xrefed_terms: str, outpath_in_terms: str, save=CONFIG['save']
+) -> Dict[str, pd.DataFrame]:
+    """Wrapper to run for all ontologies"""
+    mondo_mappings = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+
+    # 1. Xref'ed terms
+    results_xrefed: List[pd.DataFrame] = []
+    for o in ontologies:
```

Comment:
FYI: This orchestrative function iterates over ontology names, loads their files, runs analysis, and joins results.